apps/src/sites/code.org/pages/public/teacher-dashboard/index.js

```diff
@@ -366,7 +376,15 @@ function main() {
           $.each(resultStudents, function (index, student) {
             $scope.section.students.unshift(student);
           });
-        }).$promise.catch($scope.genericError);
+        }).$promise.then(() => {
+          // If we went from zero to more than zero students, rerender login
+          // type controls so the correct options are available.
+          // TODO: Once everything is React this should become unnecessary.
+          if (newStudents.length === $scope.section.students.length) {
+            unmountLoginTypeControls();
+            renderLoginTypeControls($scope.section.id);
```

Comment:
New comment:
```js
// If we started with zero students, we need to rerender login type
// controls so the correct options are available.
// Because 'temporary' students are included in $scope.section.students
// before we reach this save() action, if _all_ students are new
// students then we had zero saved students to begin with.
if (newStudents.length === $scope.section.students.length) {
```